### PR TITLE
feat(Dropdown): add split button action variant

### LINKED ISSUE #3226: Split button variation
This is a follow up React implementation to Split button variation which should be completed in Core first. The Core issue is https://github.com/patternfly/patternfly-next/issues/2326

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #3226

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md [hunk 1]
```diff
@@ -780,9 +883,129 @@ class SplitButtonDisabledDropdown extends React.Component {
...
+      <React.Fragment>
+        <Dropdown
+          onSelect={this.onActionSelect}
+          toggle={
+            <DropdownToggle
+              splitButtonItems={[
+                <DropdownToggleAction
+                  id="example-action-text"
+                  key="action"
+                  aria-label="Action button"
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md]: This button doesn't need an `aria-label` since it has text in it that should describe the action.

**Comment by kmcfaul** [comment]
@mcoker Yeah if you can open up a bug that would be great. I've removed the aria-label.

**Comment by mcoker** [comment]
@kmcfaul the expanded border has been fixed in core https://github.com/patternfly/patternfly-next/pull/2449 and is in `2.41.1`

**Comment by mcoker** [comment]
@kmcfaul https://github.com/patternfly/patternfly-next/pull/2451/ was merged in core to support adding icons to the dropdown component menu.

* Adds a modifier to menu items (`.pf-m-icon`) to use on items that have an icon in them.
* Adds a new element to the dropdown (`.pf-c-dropdown__menu-item-icon`) that will hold the icon/image/etc

File: packages/patternfly-4/react-core/src/components/Dropdown/DropdownIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import styles from '@patternfly/react-styles/css/components/Dropdown/dropdown';
+
+export interface DropdownIconProps extends React.HTMLProps<HTMLAnchorElement> {
+  /** Icon to be rendered in the dropdown item */
+  children?: React.ReactNode;
+  /** Classes applied to span element of dropdown icon item */
+  className?: string;
+}
+
+export const DropdownIcon: React.FunctionComponent<DropdownIconProps> = ({
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/DropdownIcon.tsx]: Should we call this `DropdownItemIcon`? That would align closer with what's in core and reserve `DropdownIcon` for some other thing we might add down the road.

File: packages/patternfly-4/react-core/src/components/Dropdown/DropdownItemIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  /** Classes applied to span element of dropdown icon item */
+  className?: string;
+}
+
+export const DropdownItemIcon: React.FunctionComponent<DropdownItemIconProps> = ({
+  children,
+  className = '',
+  ...props
+}: DropdownItemIconProps) => (
+  <span className={css(styles.dropdownMenuItemIcon)} {...props}>
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/DropdownItemIcon.tsx]: Should I be able to pass classes to this via `className`? Doesn't seem to work.

File: packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggleAction.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  /** Additional classes added to the DropdownToggleAction */
+  className?: string;
+  /** Flag to show if the action button is disabled */
+  isDisabled?: boolean;
+  /** A callback for when the action button is clicked */
+  onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
+  /** Element to be rendered inside the <button> */
+  children?: React.ReactNode;
+  /** Id of the action button */
+  id: string;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggleAction.tsx]: why is this required?

Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggleAction.tsx]: @tlabaj left a comment here but looks like it was removed. The `id` shouldn't be required. `id` is required on the button with the arrow (the one that toggles the menu), but for a normal action there is no requirement for an `id`.

File: packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md [hunk 5]
```diff
@@ -832,9 +936,145 @@ class SplitButtonDisabledDropdown extends React.Component {
...
+        />{' '}
+        <Dropdown
+          onSelect={this.onCogSelect}
+          toggle={
+            <DropdownToggle
+              splitButtonItems={[
+                <DropdownToggleAction
+                  id="example-action-icon"
+                  key="cog-action"
+                  aria-label="Cog icon button"
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md]: The `aria-label` text should match the text for the item in the menu, so in this case it should be `aria-label="Action"`

File: packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md [hunk 6]
```diff
@@ -832,9 +936,145 @@ class SplitButtonDisabledDropdown extends React.Component {
...
+      </DropdownItem>
+    ];
+    return (
+      <React.Fragment>
+        <Dropdown
+          onSelect={this.onActionSelect}
+          toggle={
+            <DropdownToggle
+              splitButtonItems={[
+                <DropdownToggleAction id="example-action-text" key="action" onClick={this.onActionClick}>
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md]: can you remove this `id` since it isn't needed in this example?

```suggestion
                <DropdownToggleAction key="action" onClick={this.onActionClick}>
```

File: packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md [hunk 7]
```diff
@@ -832,9 +936,145 @@ class SplitButtonDisabledDropdown extends React.Component {
...
+          isOpen={isActionOpen}
+          dropdownItems={dropdownItems}
+        />{' '}
+        <Dropdown
+          onSelect={this.onCogSelect}
+          toggle={
+            <DropdownToggle
+              splitButtonItems={[
+                <DropdownToggleAction
+                  id="example-action-icon"
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/components/Dropdown/examples/Dropdown.md]: same here - can we remove it since it isn't needed?

```suggestion
```
